City-of-Helsinki/drupal-hdbt
# UHF-7482 homecare voucher calculator

# [UHF-7482](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7482)


## What was done


* Homecare voucher codes have been updated from poc codes to real codes

## How to install

* Make sure your instance is up and running on latest dev branch.
  * `git pull origin dev`
  * `make fresh`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-7482_homecare_voucher`
* Enable calculator module in console: `drush en helfi_calculator`
* Add calculator paragraph to upper and lower content area in `/admin/structure/types/manage/page/fields`
* Enable "Home care service voucher"-calculator from `/admin/tools/calculator-settings` and add `Calculator data` from below: 
![image](https://user-images.githubusercontent.com/1191667/224933815-33231379-f5c8-4428-989f-081b52b05520.png)

<details>
<summary>CLICK ME! Calculator data for home care client payment</summary>

### Calculator data for "Home care service voucher"

```json
{
  "voucher_limits": {
    "min": 7,
    "max": 34
  },
  "household_size": {
    "1": {
      "limit": 588,
      "percent": 35,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 8,
        "9": 12,
        "13": 16,
        "17": 20,
        "21": 22,
        "25": 24,
        "29": 26,
        "33": 28,
        "37": 30,
        "41": 35
      }
    },
    "2": {
      "limit": 1084,
      "percent": 22,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 7,
        "9": 10,
        "13": 12,
        "17": 16,
        "21": 18,
        "25": 20,
        "29": 20,
        "33": 20,
        "37": 20,
        "41": 20
      }
    },
    "3": {
      "limit": 1701,
      "percent": 18,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 9,
        "13": 11,
        "17": 14,
        "21": 16,
        "25": 16,
        "29": 16,
        "33": 16,
        "37": 16,
        "41": 16
      }
    },
    "4": {
      "limit": 2103,
      "percent": 15,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 9,
        "13": 11,
        "17": 12,
        "21": 12,
        "25": 12,
        "29": 12,
        "33": 12,
        "37": 12,
        "41": 12
      }
    },
    "5": {
      "limit": 2546,
      "percent": 13,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 9,
        "13": 10,
        "17": 10,
        "21": 10,
        "25": 10,
        "29": 10,
        "33": 10,
        "37": 10,
        "41": 10
      }
    },
    "6": {
      "limit": 2924,
      "percent": 11,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 8,
        "13": 9,
        "17": 9,
        "21": 9,
        "25": 9,
        "29": 9,
        "33": 9,
        "37": 9,
        "41": 9
      }
    },
    "7": {
      "limit": 3274,
      "percent": 10,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 8,
        "13": 9,
        "17": 9,
        "21": 9,
        "25": 9,
        "29": 9,
        "33": 9,
        "37": 9,
        "41": 9
      }
    },
    "8": {
      "limit": 3624,
      "percent": 9,
      "payment_class_from_usage_per_month": {
        "0": 4,
        "5": 6,
        "9": 8,
        "13": 9,
        "17": 9,
        "21": 9,
        "25": 9,
        "29": 9,
        "33": 9,
        "37": 9,
        "41": 9
      }
    }
  },
  "beyond_cutoff": {
    "cutoff": 8,
    "limit": {
      "multiply_n": 342,
      "add": 3483
    },
    "percent": {
      "multiply_n": -1,
      "add": 9
    },
    "payment_class_from_usage_per_month": {
      "multiply_n": -1,
      "add": 0
    }
  },
  "covered_hours_per_month": 40,
  "maximum_payment_from_monthly_usage": {
    "0": 109.78,
    "6": 329.34,
    "11": 603.79,
    "21": 1152.69,
    "41": 2250.49
  },
  "voucher_divisor": 60
}
```
</details>

## How to test



* [x] Check that the calculator seems to work
    * You do not need to check the algorithm
    * The texts most likely will change after this point.

## Designers review




https://github.com/City-of-Helsinki/drupal-hdbt/pull/574

[UHF-7482]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-7482?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
